chain/client/src/info.rs

```diff
@@ -243,7 +243,13 @@ impl InfoHelper {
             for shard_id in shard_ids {
                 let mut stake_per_cp = HashMap::<ValidatorId, Balance>::new();
                 stake_sum = 0;
-                for &id in &epoch_info.chunk_producers_settlement()[shard_id as usize] {
+                let chunk_producers_settlement = &epoch_info.chunk_producers_settlement();
+                let chunk_producers = chunk_producers_settlement.get(shard_id as usize);
+                let Some(chunk_producers) = chunk_producers else {
+                    tracing::warn!(target: "stats", ?shard_id, "invalid shard id");
```

Comment:
super nit: maybe include some more info about where we got this invalid shard ID? maybe something like:

```diff
diff --git a/chain/client/src/info.rs b/chain/client/src/info.rs
index 0fc2e11c4..19919a3b0 100644
--- a/chain/client/src/info.rs
+++ b/chain/client/src/info.rs
@@ -246,7 +246,10 @@ impl InfoHelper {
                 let chunk_producers_settlement = &epoch_info.chunk_producers_settlement();
                 let chunk_producers = chunk_producers_settlement.get(shard_id as usize);
                 let Some(chunk_producers) = chunk_producers else {
-                    tracing::warn!(target: "stats", ?shard_id, "invalid shard id");
+                    tracing::warn!(
+                        target: "stats", ?shard_id, epoch_id = ?&head.epoch_id,
+                        "Invalid shard id in shard layout. Not found in epoch's chunk_producers_settlement"
+                    );
                     continue;
                 };
                 for &id in chunk_producers {
```